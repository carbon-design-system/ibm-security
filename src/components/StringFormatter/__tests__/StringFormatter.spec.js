/**
 * @file String formatter tests.
 * @copyright IBM Security 2019
 */

import { shallow } from 'enzyme';
import React from 'react';

import { StringFormatter } from '../../..';
import value from '../_mocks_';

describe('StringFormatter', () => {
  let stringFormatter;

  beforeEach(() => {
    stringFormatter = shallow(<StringFormatter value={value} />);
  });

  it('renders', () => {
    expect(stringFormatter).toMatchSnapshot();
  });

  it('renders the truncated variation', () => {
    stringFormatter.setProps({ truncate: true });

    expect(stringFormatter).toMatchSnapshot();
  });

  it('renders the correct amount of lines', () => {
    stringFormatter.setProps({ truncate: true, lines: 4 });
    expect(stringFormatter.find('span').prop('style')).toHaveProperty(
      'WebkitLineClamp',
      4
    );
  });
});
