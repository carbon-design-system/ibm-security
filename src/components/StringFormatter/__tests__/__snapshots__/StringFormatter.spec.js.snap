// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringFormatter renders 1`] = `
<span
  className="security--string-formatter"
  style={
    Object {
      "WebkitLineClamp": 1,
      "maxWidth": null,
    }
  }
>
  Buttons are used to initialize an action, either in the background or foreground of an experience. There are several kinds of buttons. Primary buttons should be used for the principle call to action on the page. Secondary buttons should be used for secondary actions on each page. Danger buttons should be used for a negative action (such as Delete) on the page. Modify the behavior of the button by changing its event properties. Small buttons may be used when there is not enough space for a regular sized button. This issue is most found in tables. Small button should have three words or less. When words are not enough, icons can be used in buttons to better communicate what the button does. Icons are always paired with text.
</span>
`;

exports[`StringFormatter renders the truncated variation 1`] = `
<TooltipDefinition
  align="start"
  className="security--string-formatter__tooltip"
  direction="bottom"
  tooltipText="Buttons are used to initialize an action, either in the background or foreground of an experience. There are several kinds of buttons. Primary buttons should be used for the principle call to action on the page. Secondary buttons should be used for secondary actions on each page. Danger buttons should be used for a negative action (such as Delete) on the page. Modify the behavior of the button by changing its event properties. Small buttons may be used when there is not enough space for a regular sized button. This issue is most found in tables. Small button should have three words or less. When words are not enough, icons can be used in buttons to better communicate what the button does. Icons are always paired with text."
>
  <span
    className="security--string-formatter security--string-formatter--truncate"
    style={
      Object {
        "WebkitLineClamp": 1,
        "maxWidth": null,
      }
    }
  >
    Buttons are used to initialize an action, either in the background or foreground of an experience. There are several kinds of buttons. Primary buttons should be used for the principle call to action on the page. Secondary buttons should be used for secondary actions on each page. Danger buttons should be used for a negative action (such as Delete) on the page. Modify the behavior of the button by changing its event properties. Small buttons may be used when there is not enough space for a regular sized button. This issue is most found in tables. Small button should have three words or less. When words are not enough, icons can be used in buttons to better communicate what the button does. Icons are always paired with text.
  </span>
</TooltipDefinition>
`;
